### PR TITLE
Support embedded relations

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Currently, if the returned JSON-LD document contains [embedded relations](https://api-platform.com/docs/core/serialization/#embedding-relations), Admin isn't able to deal with it an throws an errors.

This PR let Admin find the `@id` of the nested document to retrieve it.

### CODE REVIEW

**Comment by dunglas** [comment]
I just pushed another commit that let Admin reuse the data available in the nested document (if available) instead of sending an extra HTTP request to display the link to the relation. It dramatically improves performance.

**Comment by dunglas** [comment]
I just pushed a less intrusive solution, less hackish solution that also fixes some bugs:

* Relations to other JSON-LD documents (to-one or to-many) are always replaced by the IRI itself. It preserves the default AOR behavior.
* Every replaced relation is also stored in a cache to prevent useless extra HTTP requests to this relation.

File: src/hydra/hydraClient.js [hunk 1]
```diff
@@ -11,43 +11,89 @@ import isPlainObject from 'lodash.isplainobject';
...
-) => documents => {
-  if (!isPlainObject(documents) && !Array.isArray(documents)) {
-    return documents;
+  document,
+  clone = true,
+  addToCache = true,
+) => {
+  if (clone) {
+    // deep clone documents
+    document = JSON.parse(JSON.stringify(document));
```
Comment (soyuka) [inline src/hydra/hydraClient.js]: meh, I hope that this is never recursive or it'll break. you could use something like [fclone](https://github.com/soyuka/fclone), not sure if needed

File: src/hydra/hydraClient.js [hunk 2]
```diff
@@ -11,43 +11,89 @@ import isPlainObject from 'lodash.isplainobject';
...
+      originId: obj.id,
+      id: obj['@id'],
+    });
+  }
+
+  /**
+   * @return {string}
+   */
+  toString() {
+    return `[object ${this.id}]`;
```
Comment (soyuka) [inline src/hydra/hydraClient.js]: we could also json.stringify, not sure how this is used.

Comment (dunglas) [reply src/hydra/hydraClient.js]: It’s just a hint when debugging

File: src/hydra/hydraClient.js (hunk 1 above)
Comment (dunglas) [reply src/hydra/hydraClient.js]: It comes from a JSON, so cannot be recursive.

**Pull request merged**